src/Temporalio/Worker/TemporalWorkerOptions.cs

```diff
@@ -214,6 +214,21 @@ public TemporalWorkerOptions()
         /// </summary>
         public int MaxConcurrentActivityTaskPolls { get; set; } = 5;
 
+        /// <summary>
+        /// Gets or sets the types of exceptions that, if a workflow-thrown exception extends, will
+        /// cause the workflow/update to fail instead of suspending the workflow via task failure.
+        /// These are applied in addition to <see cref="WorkflowAttribute.FailureExceptionTypes" />
+        /// on a specific workflow. If <c>typeof(Exception)</c> is set, it effectively will fail a
+        /// workflow/update in all user exception cases.
+        /// </summary>
+        /// <remarks>
+        /// WARNING: This property is experimental and may change in the future. If unset
+        /// (i.e. left null), currently the default is to only fail the workflow/update on
+        /// <see cref="Exceptions.FailureException" /> + cancellation and suspend via task failure
+        /// all others. But this default may change in the future.
```

Comment:
```suggestion
        /// <see cref="Exceptions.FailureException" />. Other exception types will cause either cancellation or suspension via task failure. This default may change in the future.
```

Reply:
This isn't necessarily true with how the code is today. I trap cancellation errors and return workflow fail even when cancellation was not requested because of how common they appear and that it's unfair to have already-started activity cancellation be a wf fail but not-yet-started activity cancellation be a task fail.